### PR TITLE
Ensure warning only appears when there are shared roles.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #6633

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/settings/EntityOwnershipChangeNotice.js [hunk 1]
```diff
@@ -44,13 +44,15 @@ export default function EntityOwnershipChangeNotice( { slug } ) {
...
 	const storeNames = useSelect( ( select ) => {
-		const { getModuleStoreName } = select( CORE_MODULES );
+		const { getModuleStoreName, getSharedRoles } = select( CORE_MODULES );
 
 		return slugs.reduce( ( acc, currentSlug ) => {
 			const storeName = getModuleStoreName( currentSlug );
-			if ( storeName ) {
-				acc[ currentSlug ] = storeName;
+
+			if ( storeName && !! getSharedRoles( currentSlug )?.length ) {
```
Comment (aaemnnosttv) [inline assets/js/components/settings/EntityOwnershipChangeNotice.js:52]: While this works, it's a bit unnecessarily tied into the derivation of `storeNames`. Can we make it a bit more explicit that we're filtering out modules that don't have any shared roles from being included here?

File: assets/js/googlesitekit/modules/datastore/sharing-settings.js [hunk 2]
```diff
@@ -598,6 +598,7 @@ const baseSelectors = {
 			if ( sharingSettings === undefined ) {
 				return undefined;
 			}
+
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/modules/datastore/sharing-settings.js:601]: 😄

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @tofumatt – this seems to do the trick but could be a bit more clear in the way its applied. What do you think?

File: assets/js/components/settings/EntityOwnershipChangeNotice.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/components/settings/EntityOwnershipChangeNotice.js:52]: Ah, fair enough. I'll refactor it a bit 👍🏻

File: assets/js/googlesitekit/modules/datastore/sharing-settings.js (hunk 2 above)
Comment (tofumatt) [reply assets/js/googlesitekit/modules/datastore/sharing-settings.js:601]: It was just so cramped 😅

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @tofumatt 👍

**Pull request merged**